### PR TITLE
Do not use translated path

### PR DESCRIPTION
**Pull request by william-gr**
## What do these changes do?

Allow redirect to work when using url encoded paths and normalize path middleware, e.g.

/foo/bar%2Fbar will redirect to /foo/bar%2Fbar/ instead of /foo/bar/bar/

## Are there changes in behavior for the user?

Not as far as I am aware, since its a bug fix.

### CODE REVIEW

**Comment by webknjaz** [comment]
Please fill out the PR template, so that it would be easier to understand why and what you are fixing.

**Comment by asvetlov** [comment]
Sorry, I don't follow what is failed and what is expected behavior.

Unit test and changelog entry are required.

**Comment by webknjaz** [comment]
https://raw.githubusercontent.com/aio-libs/aiohttp/master/.github/PULL_REQUEST_TEMPLATE.md

**Comment by william-gr** [comment]
@asvetlov @webknjaz PR updated and summary changed to follow template.

Let me know if something is missing.

Thanks

**Comment by william-gr** [comment]
Ping. Just want to make sure there isn't anything missing here. I will be travelling the next few days and want to leave it ready for a review.

File: aiohttp/web_middlewares.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def normalize_path_middleware(
                 resolves, request = await _check_request_resolves(
                     request, path)
                 if resolves:
-                    raise redirect_class(request.path + query)
```
Comment (asvetlov) [inline aiohttp/web_middlewares.py:76]: I wonder if `request.path` can be replaced with `request.raw_path`

Comment (william-gr) [reply aiohttp/web_middlewares.py:76]: Thats the same as `path`, no? request.raw_path is being used above to populate paths_to_check.

Comment (asvetlov) [reply aiohttp/web_middlewares.py:76]: No. It is the percent encoded *canonical* path representation.

Comment (william-gr) [reply aiohttp/web_middlewares.py:76]: And thats exactly the issue I am solving here, and its passing all the tests so I dont get it.

**Comment by william-gr** [comment]
@asvetlov call updated. Thanks for the helpful explanation as of why.

**Comment by asvetlov** [comment]
thanks

**Pull request merged**